### PR TITLE
Initial Parser class and spec

### PR DESCRIPTION
**Pull request by benichu**

### CODE REVIEW

File: lib/checklister/parser.rb [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  class Parser
+    # Params :
+    # - file : the path of the markdown file to parse
+    def initialize(file_path)
+      @file_content = File.open(file_path).read
+    end
+
+    # What should the issue title be ?
+    # The plain text headline of the markdown document
+    # Returns a string
```
Comment (benichu) [inline lib/checklister/parser.rb]: Don't forget to use the [yardoc format](http://yardoc.org/)

File: spec/checklister/parser_spec.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  describe "#parse_title" do
+    it "should return a parsed title" do
+      expect(parser.parse_title).to eq("Simple Checklist")
+    end
+  end
+
+  describe "#parse_body" do
+    it "should return a parsed body" do
+      expect(parser.parse_body).to eq("When you need to accomplish stuff, follow those steps.\n- [ ] Step 1\n- [ ] Step 2\n- [ ] Step 3")
```
Comment (mdeloupy) [inline spec/checklister/parser_spec.rb]: @benichu Should I take this comment into consideration ?

Comment (benichu) [inline spec/checklister/parser_spec.rb]: I don't think so... I also have a lot of those warnings and sometimes it's not worth the effort...

File: lib/checklister/parser.rb [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    #
+    # @return [String] the parsed body
+    def parse_body
+      context = get_context
+      checklist = get_checklist
+      "#{context}\n#{checklist}"
+    end
+
+    # @return [Hash] the issue params (title & body)
+    def issue_params
```
Comment (benichu) [inline lib/checklister/parser.rb]: I am not sure about that method name. Let's talk about it tomorrow.

File: lib/checklister/parser.rb [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  class Parser
+    # @param [String] file_path the path of the markdown file to parse
+    def initialize(file_path)
+      @file_content = File.open(file_path).read
+    end
+
+    # The issue title should be the plain text headline of the markdown document
+    #
+    # @return [String] the parsed title
+    def parse_title
```
Comment (benichu) [inline lib/checklister/parser.rb]: It should not be public... Let's talk about it

File: lib/checklister/parser.rb [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+          return line.sub("# ", "").sub("\n", "")
+          break
+        end
+      end
+    end
+
+    # The issue body is composed of the parsed context (if any) and the checklist
+    #
+    # @return [String] the parsed body
+    def parse_body
```
Comment (benichu) [inline lib/checklister/parser.rb]: It should not be public

File: lib/checklister/parser.rb (hunk 4 above)
Comment (mdeloupy) [inline lib/checklister/parser.rb]: You're right, only `issue_params` should be public. I'll change that right away.

File: lib/checklister/parser.rb (hunk 3 above)
Comment (benichu) [inline lib/checklister/parser.rb]: What about `to_params`?
This way it is more generic and that class does not imply a knowledge of 'issues'...

Comment (mdeloupy) [inline lib/checklister/parser.rb]: That’s way better, I just changed it.


—
Manon Deloupy

**Pull request merged**